draft-js-side-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -36,13 +36,21 @@ export default class Toolbar extends React.Component {
     // Note: need to wait on tick to make sure the DOM node has been create by Draft.js
     setTimeout(() => {
       const node = document.querySelectorAll(`[data-offset-key="${offsetKey}"]`)[0];
-      const top = node.getBoundingClientRect().top;
-      const editor = this.props.store.getItem('getEditorRef')().refs.editor;
-      const scrollY = window.scrollY == null ? window.pageYOffset : window.scrollY;
+
+      // The editor root should be two levels above the node from
+      // `getEditorRef`. In case this changes in the future, we
+      // attempt to find the node dynamically by traversing upwards.
+      const editorRef = this.props.store.getItem('getEditorRef')();
+      if (!editorRef) return;
+      let editorRoot = editorRef.refs.editor;
+      while (editorRoot.className.indexOf('DraftEditor-root') === -1) {
+        editorRoot = editorRoot.parentNode;
+      }
+
       this.setState({
         position: {
-          top: (top + scrollY),
-          left: editor.getBoundingClientRect().left - 80,
+          top: node.offsetTop + editorRoot.offsetTop,
+          left: editorRoot.offsetLeft - 80,
```

Comment:
ok I just tested this and this now seems to break when the root node isn't relatively positioned.

Mind fixing that (I guess you just need to treat the positioning differently if a position is set)

Also - if you could add a story for relatively positioned ones that'd be fab.

Reply:
I think you're referring to the existing stories for the side toolbar – is that correct? It was already working in the docs, so I was a bit confused. I've seen that the Draft.css stylesheet wasn't loaded in the Storybook. I've added that import and a story and now everything seems fine.

Reply:
it was however working before (non positioned and without the draft.css stylesheet)?

Reply:
Yeah, that's true. It's because the positioning strategy is different now. I think it can be considered a bug though if Draft.css is missing in an app. From the Draft.js docs:
> The Draft framework includes a handful of CSS resources intended for use with the editor, available in a single file via the build, Draft.css.

> This CSS should be included when rendering the editor, as these styles set defaults for text alignment, spacing, and other important features. Without it, you may encounter issues with block positioning, alignment, and cursor behavior.

> If you choose to write your own CSS independent of Draft.css, you will most likely need to replicate much of the default styling.

([source](https://draftjs.org/docs/advanced-topics-issues-and-pitfalls.html#missing-draft-css))

We're also [mentioning this in our FAQ](https://github.com/draft-js-plugins/draft-js-plugins/blob/master/FAQ.md#my-default-styles-are-broken).

Reply:
ok I have it working now o_O . this is awesome ♥️ !!!!

the only thing now is that when we don't include Draft.css it looks like this:

![image](https://user-images.githubusercontent.com/1188186/39199757-2436a3fc-47e2-11e8-8e72-2bb9cd3e775c.png)

Bear in mind that draft.css actually adds position:relative to the root...

Reply:
Yeah, right – you really have to include the stylesheet 🙂. It's that `position: relative` on the editor root that is necessary for the positioning to work correctly (also every other kind of position will do, as long as it isn't `static`). The PR fixes the use case where theres an enclosing parent that is positioned relatively.

Reply:
hmhmhm ok! We just need to make that extra clear imo!

Reply:
I'll open an issue to add this to the docs